source/common/http/http2/codec_impl.cc

```diff
@@ -853,6 +853,8 @@ ConnectionImpl::Http2Options::Http2Options(const Http2Settings& http2_settings)
   // of kept alive HTTP/2 connections.
   nghttp2_option_set_no_closed_streams(options_, 1);
   nghttp2_option_set_no_auto_window_update(options_, 1);
+  nghttp2_option_set_max_send_header_block_length(options_,
```

Comment:
Isn't this the setting for the size of an individual header frame?  I think we want to change the total allowed headers, not the per-frame limits.

Reply:
IIRC this is not very well named, but I think this does actually set the header send limit, and the library will chunk it into frames using continuation frames if necessary. @auni53 to check me here.

Reply:
Could be.  I'm not seeing any other size-related fields to tweak.  I'd checked
https://nghttp2.org/documentation/nghttp2_option_set_max_send_header_block_length.html
"This option sets the maximum length of header block (a set of header fields per one HEADERS frame) to send. "
It was the "per one HEADERS frame" that got me thinking it might be the wrong field.
in code it's used in prep_frame which I thought was after the chunking but I could easily be misremembering.  I'm happy to let Auni do the code sleuthing though :-)

Reply:
Yeeee, I'm pretty certain this is the right call. Also because I did this TDD-style and the testing behaviour checks out.